proxy/http2/Http2Stream.cc

```diff
@@ -259,13 +259,16 @@ Http2Stream::send_request(Http2ConnectionState &cstate)
     return;
   }
 
-  if (this->recv_end_stream) {
-    this->read_vio.nbytes = bufindex;
-    this->signal_read_event(VC_EVENT_READ_COMPLETE);
-  } else {
-    // End of header but not end of stream, must have some body frames coming
-    this->has_body = true;
-    this->signal_read_event(VC_EVENT_READ_READY);
+  // Is the _sm ready to process the header?
+  if (this->read_vio.nbytes > 0) {
```

Comment:
When `read_vio.nbytes == 0`, ATS retry thie `Http2Stream::send_request()` call by somehow?

Reply:
NVM. Retry on the HTTP/2 side is not required. My understanding is 
- If HttpSM can't get hook continuation lock in above, it schedule `EVENT_INTERVAL`. 
- When HttpSM handles `EVENT_INTERVAL`, it'll read request in this read_vio.